### ISSUE TITLE
Tranmission 2.93 compatability

### ISSUE THREAD

**Issue by Tangbravo221** [transmission-remote-gui/transgui#1099]

**Comment by antekgla** [comment]
I don´t understand...
In 2.93 it is possible authenticate with username only (**without password**)
or without username/password at all?

**Comment by AffekeNommu** [comment]
I am also having the same issue. I can get to my copy of Transmission 2.93 via the web interface with password and no username just fine but I cannot configure Transmission Remote GUI to get to it as it does not save the login credentials if the username is blank.

**Comment by Tangbravo221** [comment]
It’s possible to authenticate without USERNAME, PW is still required.

T-GUI 5.14 won’t allow any sort of authentication without a Username.

It’s either all or nothing on authentication.

**Comment by teufelhunden0311** [comment]
It seems the remote gui is busted until an update allows authentication without a username.

**Comment by Tangbravo221** [comment]
Yup, but I hadn’t see anything on it so I thought I would post it.

**Comment by antekgla** [comment]
I try to look at it...
...but I dont understand: what is the advantage of not have a Username and only password?
What security problem resolve that?
There any scenario where that is desirable?

**Comment by Tangbravo221** [comment]
No idea of the advantage. You would have to ask the authors of Transmission 2.93…

It’s caused a lot of problems not just to users, but to seedbox support…

**Comment by leonsoft-kras** [comment]
@antekgla - there is version 2.93 for windows. We need to install and configure the input of only the password.  Next, look under the debugger, what happens in the TRGUI. 
Perhaps the problem isn't only in the TRGUI. It may be necessary to modify the RPC protocol in the program.

**Comment by Tangbravo221** [comment]
This is what my provider said was needed:

 

For things which require a username it is possible to add one to the rpc-username field of ~/.config/transmission-daemon/settings.json. 

Please bear in mind you'll need to stop the Transmission process prior to making changes. Also, adding a username will negatively impact the web UI - it will continue to auto-refresh until you press enter on the URL bar.

**Comment by antekgla** [comment]
@Tangbravo221 you need to do what your provider said:

- Log in your seedbox (using putty or another terminal program)
- Stop the transmission daemon (ask your provider how do that)
- Edit ~/.config/transmission-daemon/settings.json adding in the  rpc-username the username you want (between quotes)
- Save ~/.config/transmission-daemon/settings.json
- Start the transmission daemon again.
- Log in with TransGUI with the new username.

**Comment by Tangbravo221** [comment]
Actually I’m just going to switch to deluge and be done with it, especially if T-GUI will not be updated.

**Comment by PeterDaveHello** [comment]
@Tangbravo221 no offense, but I think maybe less emotions here can help solving the problems, we're all trying to imrpove this project by our own spare time.

**Comment by AffekeNommu** [comment]
I have been having a poke around the code and can see that blank username is being used to set the authentication flag in LoadConnSettings in connoptions.pas.
I have added a username in the rpc-username settings in the json and login is working happily for Transmission Remote GUI now.
Really do not know why Transmission chose to default to password and blank username...

**Comment by antekgla** [comment]
> I have been having a poke around the code and can see that blank username is being used to set the authentication flag in LoadConnSettings in connoptions.pas.

Yes I see it but that is not the problem. That line in LoadConnSettings in connoptions.pas only prevent of read the password from ini file if the username is blank and fill the password edit.

I have being debugging and seems a deeper problem. I guess what is not transGUI fault besides a problem in synapse library.., but I am only guessing and I need to investigate a lot more.

**Comment by Tangbravo221** [comment]
I’m sorry, I had no idea I was showing emotion. I get this a time contribution thing and that it’s fixing something done by the authors of Transmission. I just meant to say that for me, trying to change a file and going in via SSH is right on the edge of the cliff for my expertise. I’m offered rtorrent and deluge and rather than overtax my limited skills, I’m better off going temporarily to another platform.

 

I take no offense and my apologies if someone took offense at what I wrote. It was not, in any way meant to critize or offend anyone (except maybe ok, whoever made this change at transmission and forcing it (as opposed to allowing the old way to continue.)

**Comment by antekgla** [comment]
Fixed in https://github.com/transmission-remote-gui/transgui/commit/fafa368c84bd7751913609223ba8d328592ade5b

Was an error in Synapse library what exclude all authentication info when the user is empy.
Now checks user AND pass, if any is not empty the auth info is included.
Also adapted the **LoadConnSettings** procedure in connoptions.pas to load the password also when the user is empty.

@Tangbravo221 sadly you have to wait to a new release to enjoy this fixes :cry: unless what @PeterDaveHello make a 5.15.1 release :wink:

**Comment by PeterDaveHello** [comment]
@antekgla not really, we have temporarily builds in every commit :)

**Comment by Tangbravo221** [comment]
Wouldn’t you know it, I’m on windows 10 x64….

**Comment by PeterDaveHello** [comment]
@Tangbravo221 hmmm, I'll release v5.15.1 soon for that, as we don't have automatic build for Windows, it takes more time and manual works to release for it.

**Comment by Tangbravo221** [comment]
No problem I appreciate it.  Amazing this thing you guys do….Most folks have no idea….

**Comment by PeterDaveHello** [comment]
Thanks. Hopefully, v5.15.1 will be released within 12 hours.

**Comment by Tangbravo221** [comment]
Wow, on behalf of so many that wouldn’t know who to thank…THANK YOU!

**Comment by PeterDaveHello** [comment]
@Tangbravo221 due to some reasons, I have to push the release a little bit later, about 6 hours later. Just let you know.

**Comment by Tangbravo221** [comment]
No problem. Appreciate your time and effort. Thank you! (and for the update also!)

**Comment by PeterDaveHello** [comment]
@Tangbravo221 you're welcome

**Comment by cfpp2p** [comment]
@antekgla @PeterDaveHello 
daemon also accepts empty for both in settings.json and authenticates with web client.
I can't authenticate like that with transgui right now.
I don't know what's a use case or security reason but regardless, whether we like this daemon behavior or not.
```
    "rpc-authentication-required": true, 
    "rpc-password": "", 
    "rpc-username": "", 
```

https://github.com/transmission-remote-gui/transgui/commit/fafa368c84bd7751913609223ba8d328592ade5b
looks like we can do entirely without `cbAuth.Checked` and `if (User <> '') or (Pass <> '') then`

All other usename password configurations function with v5.15.2

thanks for v5.15.2 :)

**Comment by PeterDaveHello** [comment]
@cfpp2p I saw that you have a forked version of transgui, maybe you'd like to contribution some codes? :)

**Comment by cfpp2p** [comment]
@PeterDaveHello thanks for opportunity, we'll see, time is crazy...

**Comment by PeterDaveHello** [comment]
@cfpp2p 😄

**Comment by barbequesauce** [comment]
@cfpp2p described my use case nicely. No username/password, instead access is controlled by the rpc whitelist. Hoping this can be addressed...

**Comment by antekgla** [comment]
@cfpp2p @barbequesauce 
What is the point of have **Authentication = true** and no username and password. Makes no sense...
Make **Authentication= false** and TransGUI would be connect leaving Authentication checkmark uncheck.

**Comment by barbequesauce** [comment]
My understanding was that rpc-authentication-required must be true to enable the whitelists...

**Comment by antekgla** [comment]
`My understanding was that rpc-authentication-required must be true to enable the whitelists...`

It not true... You can make **Authentication=false** and have Whitelist active.
Whitelist on/off is controlled by **rpc-whitelist-enabled = true**
